### PR TITLE
deploy: avoid replacing socat image

### PR DESCRIPTION
**Pull request by pohly**
Setting IMAGE_TAG or IMAGE_REGISTRY unintentionally also affected the
socat image. This also disables SOCAT_IMAGE_TAG/REGISTRY, but that
should not be needed and wasn't documented either.

### CODE REVIEW

**Comment by pohly** [comment]
/hold

Still needs to be tested.

**Comment by pohly** [comment]
/hold cancel

Here's a succesful test run with this change: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-csi-canary-on-kubernetes-master/1114802533932470277

File: deploy/util/deploy-hostpath.sh [hunk 1]
```diff
@@ -105,7 +105,7 @@ for i in $(ls ${BASE_DIR}/hostpath/*.yaml | sort); do
     echo "   $i"
     modified="$(cat "$i" | while IFS= read -r line; do
         nocomments="$(echo "$line" | sed -e 's/ *#.*$//')"
-        if echo "$nocomments" | grep -q '^\s*image:\s*'; then
+        if echo "$nocomments" | grep -q '^\s*image:\s*' && ! echo "$nocomments" | grep -q image:.*/socat; then
```
Comment (msau42) [inline deploy/util/deploy-hostpath.sh]: Can we use a variable for blacklisted components?

**Pull request merged**